feature_engine/scaling/mean_normalization.py

```diff
@@ -0,0 +1,186 @@
+
+    If a column is constant, then it will be transformed to the zero constant column.
+
+    Parameters
+    ----------
+    {variables}
+
+
+    Attributes
+    ----------
+    params_:
+        a dictionary containing the mean, max and min of every given variable
+
+    {variables_}
+
+    {feature_names_in_}
+
+    {n_features_in_}
+
+    Methods
+    -------
+    {fit}
+
+    {fit_transform}
+
+    {inverse_transform}
+
+    transform:
+        Scale the variables using mean normalization.
+
+    Examples
+    --------
+
+    >>> import numpy as np
+    >>> import pandas as pd
+    >>> from feature_engine.scaling import MeanNormalizationScaler
+    >>> np.random.seed(42)
+    >>> X = pd.DataFrame(dict(x = np.random.lognormal(size = 100)))
+    >>> mns = MeanNormalizationScaler()
+    >>> mns.fit(X)
+    >>> X = mns.transform(X)
+    >>> X.head()
+            x
+    0  0.496714
+    1 -0.138264
+    2  0.647689
+    3  1.523030
+    4 -0.234153
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+    ) -> None:
+
+        self.variables = _check_variables_input_value(variables)
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        The method populate the variable `params_`.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features].
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y: pandas Series, default=None
+            It is not needed in this transformer. You can pass y or None.
+        """
+
+        # check input dataframe
+        X = super().fit(X)
+        self.params_ = {
+            "mean": X[self.variables_].mean(),
+            "range": X[self.variables_].max() - X[self.variables_].min(),
+        }
+
+        # check for constant columns
+        range_columns = self.params_["range"]
+        constant_columns = range_columns[range_columns == 0].index.tolist()
+        if constant_columns:
+            raise ValueError(
+                "Division by zero in the scaling. This is because \n"
+                f"the following column/s are constant: {constant_columns}"
+            )
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transform the variables using mean normalization.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features]
+            The data to be transformed.
+
+        Returns
+        -------
+        X_new: pandas dataframe
+            The dataframe with the transformed variables.
+        """
+
+        # check input dataframe and if class was fitted
+        X = self._check_transform_input_and_state(X)
+
+        # transformation
+        X[self.variables_] = (
+            X[self.variables_] - self.params_["mean"][self.variables_]
```

Comment:
is the self.variables_  in `self.params_["mean"][self.variables_]`  necessary?

Reply:
I think yes. In this way you are taking on both side the columns in the right order

Reply:
but I think that we can also do it with numpy array actually